code/modules/mob/living/carbon/carbon.dm

```diff
@@ -251,17 +251,40 @@
 					"<span class='notice'>You pat [src] on the head.</span>",\
 					)
 				else
-
-					M.visible_message(\
-					"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
-					"<span class='notice'>You hug [src].</span>",\
-					)
-					if(ishuman(src))
-						var/mob/living/carbon/human/H = src
-						if(H.wear_suit)
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+					if(M.zone_selected == "r_arm")
```

Comment:
```suggestion
					if(M.zone_selected == ("r_arm" || "l_arm" || "r_hand" || "l_hand"))
```
you can simplify all bodypart reactions by doing this

Reply:
Thanks, I hadn't thought of it that way. I had another way to place the "handshakes" but I don't know if it would work in the same way